### PR TITLE
feat(containers): Adds ShortcutManager

### PR DESCRIPTION
**Pull request by talend-glorieux**
**What is the problem this PR is trying to solve?**
We don't have a common way to handle keyboard shortcuts.

**What is the chosen solution to this problem?**
Add a ShortcutManager container that handles keyboard shortcuts.
It takes a map of keys to routes and/or actionCreator.

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: packages/containers/src/ShortcutManager/__snapshots__/ShortcutManager.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Container ShortcutManager should render 1`] = `Object {}`;
```
Comment (jmfrancois) [inline packages/containers/src/ShortcutManager/__snapshots__/ShortcutManager.test.js.snap]: too weird to be a snapshots convert it to normal tests please

File: packages/containers/src/index.js [hunk 2]
```diff
@@ -23,9 +23,12 @@ import ObjectViewer from './ObjectViewer';
 import Redirect from './Redirect';
 import SidePanel from './SidePanel';
```
Comment (jmfrancois) [inline packages/containers/src/index.js]: useless empty line. please remove

File: packages/containers/src/index.js [hunk 3]
```diff
@@ -23,9 +23,12 @@ import ObjectViewer from './ObjectViewer';
...
 
+import ShortcutManager from './ShortcutManager';
+
 import actionAPI from './actionAPI';
 import componentState from './state';
 
+
```
Comment (jmfrancois) [inline packages/containers/src/index.js]: same here

File: packages/containers/src/index.js [hunk 4]
```diff
@@ -52,4 +55,6 @@ export {
 	Layout,
 	TooltipTrigger,
 	Typeahead,
+
+	ShortcutManager,
 };
```
Comment (jmfrancois) [inline packages/containers/src/index.js]: please make it alphabetic ordered (import + export)

File: output/containers.eslint.txt [hunk 5]
```diff
@@ -35,11 +35,18 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   1:8   error  'React' is defined but never used                                                              no-unused-vars
   7:18  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
 
+/home/travis/build/Talend/ui/packages/containers/src/ShortcutManager/ShortcutManager.container.js
+  2:1  error  'keycode' should be listed in the project's dependencies. Run 'npm i -S keycode' to add it  import/no-extraneous-dependencies
+
+/home/travis/build/Talend/ui/packages/containers/src/ShortcutManager/ShortcutManager.test.js
+  2:1  error  'keycode' should be listed in the project's dependencies. Run 'npm i -S keycode' to add it  import/no-extraneous-dependencies
```
Comment (jmfrancois) [inline output/containers.eslint.txt]: please add keycode as peerDeps + devDeps

File: packages/containers/src/ShortcutManager/ShortcutManager.container.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	// eslint-disable-next-line class-methods-use-this
+	componentWillUnmount() {
+		document.removeEventListener('keydown');
+	}
+
+	redirect(event, path) {
+		this.props.dispatchActionCreator('redirect', event, { action: { path } });
+	}
+
+	handleRegexMatching(event, redirectMap, pathname) {
```
Comment (jmfrancois) [inline packages/containers/src/ShortcutManager/ShortcutManager.container.js]: this one is not unittested

File: packages/containers/src/ShortcutManager/ShortcutManager.container.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	componentDidMount() {
+		document.addEventListener('keydown', this.handleKeyPress);
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	componentWillUnmount() {
+		document.removeEventListener('keydown');
+	}
```
Comment (jmfrancois) [inline packages/containers/src/ShortcutManager/ShortcutManager.container.js]: if this is possible (to mock document) I would like to see unittest on didMount and WillUnmout.

Comment (talend-glorieux) [inline packages/containers/src/ShortcutManager/ShortcutManager.container.js]: You can mock the document globally. Breaking it for all tests :(

File: packages/containers/src/ShortcutManager/ShortcutManager.container.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+		this.handleRegexMatching = this.handleRegexMatching.bind(this);
+	}
+
+	componentDidMount() {
+		document.addEventListener('keydown', this.handleKeyPress);
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	componentWillUnmount() {
+		document.removeEventListener('keydown');
```
Comment (jare-talend) [inline packages/containers/src/ShortcutManager/ShortcutManager.container.js]: To remove the listener, you need to add the listener else he fails. The removeListener needs to match to the addEventListener.

Comment (jmfrancois) [inline packages/containers/src/ShortcutManager/ShortcutManager.container.js]: nice catch: https://developer.mozilla.org/fr/docs/Web/API/EventTarget/removeEventListener

Comment (talend-glorieux) [inline packages/containers/src/ShortcutManager/ShortcutManager.container.js]: Nice catch indeed :)

**Pull request merged**